packages/backend/src/functions/httpTrigger.ts

```diff
@@ -130,10 +131,14 @@ async function getProvenance(request: HttpRequest, context: InvocationContext):
         const { data, timestamp } = await decryptBlob(blobClient, deviceKey);
         const json = new TextDecoder().decode(data);
         const provRecord = JSON.parse(json) as ProvenanceRecord;
+        const deviceID = await calculateDeviceID(deviceKey);

```

Comment:
Why are we calculating the device key again? It's calculated at the top of the function. And why do we need the device id in each provenance record? I could see adding it once at the top level, but not repeating it in every record

Reply:
I think you are right. I have removed this.